tools/utils/utils.js

```diff
@@ -472,21 +479,21 @@ exports.isPathRelative = function (x) {
 //
 // This is talking about NPM/Cordova versions specifically, not Meteor versions.
 // It does not support the wrap number syntax.
-exports.ensureOnlyExactVersions = function (dependencies) {
+exports.ensureOnlyExactVersions = function (dependencies, {forNpm}) {
   _.each(dependencies, function (version, name) {
     // We want a given version of a smart package (package.js +
     // .npm/npm-shrinkwrap.json) to pin down its dependencies precisely, so we
     // don't want anything too vague. For now, we support semvers and urls that
     // name a specific commit by SHA.
-    if (! exports.isExactVersion(version)) {
+    if (! exports.isExactVersion(version, {forNpm})) {
       throw new Error(
         "Must declare exact version of dependency: " + name + '@' + version);
     }
   });
 };
-exports.isExactVersion = function (version) {
-  return semver.valid(version) || exports.isUrlWithSha(version)
-    || exports.isUrlWithFileScheme(version);
+exports.isExactVersion = function (version, {forNpm}) {
```

Comment:
This may be bikeshedding, but would it make sense to turn the flag around and use `forCordova` instead of `forNpm`? Strictly taken, Cordova also uses NPM, so ultimately it is a URL to be passed to NPM.